server/controllers/profile.js

```diff
@@ -0,0 +1,77 @@
+const asyncHandler = require('express-async-handler');
+const Profile = require('../models/Profile');
+
+// @desc Get all profiles
+// @route GET /profiles
+// @access Public
+exports.getProfiles = asyncHandler(async (req, res, next) => {
+  const profiles = await Profile.find();
+
+  res
+    .status(200)
+    .json({ success: true, data: profiles, count: profiles.length });
+});
+
+// @desc Get single profile
+// @route GET /profiles/:id
+// @access Public
+exports.getProfile = asyncHandler(async (req, res, next) => {
+  const profile = await Profile.findById(req.params.id);
```

Comment:
Optional: You can validate the path parameters here

```
 if (!mongoose.Types.ObjectId.isValid(req.params.id)) {
    return res.status(400).send("Bad Request");
  }
```

Reply:
I validated the params with this in the middleware/error.js file

if(err.name === 'CastError') {
    err.message = `No resource foound for id ${err.value}`;
    statusCode = 404;
  }